distros/generate.sh

```diff
@@ -9,7 +9,13 @@ set -e
 #	or: ./generate.sh debian-newversion
 #		to create a new folder and a Dockerfile within it
 
-cd "$(dirname "$(readlink -f "$BASH_SOURCE")")"
+# `readlink -f` does not work on macOS.
+# To get `greadlink`, run `brew install coreutils`.
+if [ "$(uname)" == "Darwin" ]; then
```

Comment:
Ok, so it turns out the existing readlink option only supports one layer of indirection via symlink, and only if the generate.sh script itself is an absolute (not relative) symlink.

Since we're not symlinking the generate.sh script, I think we can completely do away with the readlink and go with something simpler, like 
```
cd "$(dirname "${BASH_SOURCE[0]}")"
```